adapters/flipp/flipp.go

```diff
@@ -0,0 +1,219 @@
+package flipp
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/buger/jsonparser"
+	"github.com/gofrs/uuid"
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/prebid/prebid-server/util/httputil"
+)
+
+const (
+	bannerType      = "banner"
+	inlineDivName   = "inline"
+	flippBidder     = "flipp"
+	defaultCurrency = "USD"
+)
+
+var (
+	count    int64 = 1
+	adTypes        = []int64{4309, 641}
+	dtxTypes       = []int64{5061}
+)
+
+type adapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the Flipp adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	adapterRequests := make([]*adapters.RequestData, 0, len(request.Imp))
+	var errors []error
+
+	for _, imp := range request.Imp {
+		adapterReq, err := a.processImp(request, imp)
+		if err != nil {
+			errors = append(errors, err)
+		}
+
+		if adapterReq != nil {
+			adapterRequests = append(adapterRequests, adapterReq)
+		}
+	}
+	return adapterRequests, errors
+}
+
+func (a *adapter) makeRequest(request *openrtb2.BidRequest, campaignRequestBody CampaignRequestBody) (*adapters.RequestData, error) {
+	campaignRequestBodyJSON, err := json.Marshal(campaignRequestBody)
+	if err != nil {
+		return nil, err
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json")
+	headers.Add("User-Agent", request.Device.UA)
+	return &adapters.RequestData{
+		Method:  http.MethodPost,
+		Uri:     a.endpoint,
+		Body:    campaignRequestBodyJSON,
+		Headers: headers,
+	}, err
+}
+
+func (a *adapter) processImp(request *openrtb2.BidRequest, imp openrtb2.Imp) (*adapters.RequestData, error) {
+	var flippExtParams openrtb_ext.ImpExtFlipp
+	params, _, _, err := jsonparser.Get(imp.Ext, "bidder")
+	if err != nil {
+		return nil, fmt.Errorf("flipp params not found. %v", err)
+	}
+	err = json.Unmarshal(params, &flippExtParams)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to extract flipp params. %v", err)
+	}
```

Comment:
could use `json.Unmarshal` instead of `jsonparser.Get`

```
var impExt adapters.ExtImpBidder
if err := json.Unmarshal(imp.Ext, &impExt); err != nil {
	return nil, fmt.Errorf("error parsing imp.Ext %v", err)
}

var flippExtParams openrtb_ext.ImpExtFlipp
if err := json.Unmarshal(impExt.Bidder, &flippExtParams); err != nil {
	return nil, fmt.Errorf("Unable to extract flipp params. %v", err)
}
```